src/main/java/uk/nhs/hee/trainee/details/dto/CctCalculationDetailDto.java

```diff
@@ -94,7 +94,10 @@ public record CctProgrammeMembershipDto(
 
       @NotNull
       @Range(min = 0, max = 1)
-      Double wte) {
+      Double wte,
+
+      @NotNull
+      String designatedBodyCode) {
```

Comment:
@DorisWongNHS are you sure we'll always have a designated body code (or will it be set to "" if we don't?)

I did a quick check on TCS with
```
SELECT p.status, count(*) FROM tcs.Programme p 
LEFT JOIN reference.LocalOffice lo ON p.owner = lo.name
LEFT JOIN reference.DBC dbc ON lo.abbreviation = dbc.abbr
WHERE dbc.dbc IS NULL
group by p.status
```
and there are +16k programmes without DBC: these are almost all INACTIVE, but there were 4 without a status and 1 marked DELETE. Which seems ok, but just checking you are comfortable with the data 😁

Reply:
(In the front-end PR it seems designatedBodyCode is allowed to be null/missing)

Reply:
Yeah, I was also thinking about it yesterday. Aware that DBC should be exist in all PM and is needed in the LTFT on Admins-UI, With the setting NotNull, error will prompt in the FE if the DBC is missing when save, so we can know about the special case and take further action.